torch/_refs/__init__.py

```diff
@@ -1822,6 +1824,73 @@ def tensor_split(
         return tuple(splits)
 
 
+def hsplit(
+    a: TensorLikeType, indices_or_sections: DimsType
+) -> Tuple[TensorLikeType, ...]:
+    if a.ndim < 1:
+        msg = (
+            "torch.hsplit requires a tensor with at least 1 dimension, but got a tensor with "
+            + str(a.ndim)
+            + " dimensions!"
+        )
+        raise RuntimeError(msg)
+    dim = 0 if a.ndim == 1 else 1
+    if isinstance(indices_or_sections, int):
+        split_size = indices_or_sections
+        if not (split_size != 0 and a.shape[dim] % split_size == 0):
+            msg = (
+                "torch.hsplit attempted to split along dimension "
+                + str(dim)
+                + ", but the size of the dimension "
+                + str(a.shape[dim])
+                + " is not divisible by the split_size "
+                + str(split_size)
+                + "!"
+            )
+            raise RuntimeError()
+        return tensor_split(a, split_size, dim)
+    elif isinstance(indices_or_sections, (list, tuple)):
+        split_sizes = indices_or_sections
+        return tensor_split(a, split_sizes, dim)
+    else:
```

Comment:
No need for the else clause, this can just fall through to the error

Even nicer, for style points, is you can do:

```
if (int):

if not isinstance(indices_or_sections, (list, tuple)):
  error

fallthrough to case where it's a list or tuple
```

that would minimize indentation
```